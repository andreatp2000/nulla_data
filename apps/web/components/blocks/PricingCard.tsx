interface PricingCardProps {
  title: string
  price: string
  features: string[]
  highlight?: boolean
}

const buttonClasses =
  'rounded bg-nd-primary py-2 text-black hover:bg-nd-secondary focus-visible:outline-none focus-visible:' +
  'ring-2 focus-visible:ring-nd-ring'

export function PricingCard({
  title,
  price,
  features,
  highlight,
}: PricingCardProps) {
  return (
    <div
      className={`rounded-lg border p-6 ${highlight ? 'border-nd-primary' : 'border-nd-surface'} flex flex-col`}
    >
      <h3 className="mb-2 text-xl font-semibold">{title}</h3>
      <p className="mb-4 text-3xl font-bold">{price}</p>
      <ul className="mb-6 flex-1 space-y-2 text-sm">
        {features.map((f) => (
          <li key={f}>â€¢ {f}</li>
        ))}
      </ul>
      <button className={buttonClasses}>Scegli</button>
    </div>
  )
}
