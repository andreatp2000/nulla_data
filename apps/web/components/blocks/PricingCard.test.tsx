import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { PricingCard } from './PricingCard'

import { test, expect } from 'vitest'
=======


test('pricing card snapshot', () => {
  const { container } = render(
    <PricingCard title="Test" price="10â‚¬" features={['a', 'b']} />
  )
  expect(container).toMatchSnapshot()
})
