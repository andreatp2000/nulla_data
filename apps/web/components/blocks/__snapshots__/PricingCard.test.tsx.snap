// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pricing card snapshot 1`] = `
<div>
  <div
    class="rounded-lg border p-6 border-nd-surface flex flex-col"
  >
    <h3
      class="mb-2 text-xl font-semibold"
    >
      Test
    </h3>
    <p
      class="mb-4 text-3xl font-bold"
    >
      10€
    </p>
    <ul
      class="mb-6 flex-1 space-y-2 text-sm"
    >
      <li>
        • 
        a
      </li>
      <li>
        • 
        b
      </li>
    </ul>
    <button
      class="rounded bg-nd-primary py-2 text-black hover:bg-nd-secondary focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-nd-ring"
    >
      Scegli
    </button>
  </div>
</div>
`;
